src/v/kafka/requests/fetch_request.cc

```diff
@@ -563,4 +563,70 @@ fetch_api::process(request_context&& rctx, ss::smp_service_group ssg) {
     });
 }
 
+void op_context::reset_context() {
+    initial_fetch = false;
+    response_iterator = response.begin();
+}
+
+// decode request and initialize budgets
+op_context::op_context(request_context&& ctx, ss::smp_service_group ssg)
+  : rctx(std::move(ctx))
+  , ssg(ssg)
+  , response_size(0)
+  , response_error(false)
+  , response_iterator(response.begin()) {
+    /*
+     * decode request and prepare the inital response
+     */
+    request.decode(rctx);
+    if (likely(!request.topics.empty())) {
+        response.partitions.reserve(request.topics.size());
+    }
+
+    if (auto delay = request.debounce_delay(); delay) {
```

Comment:
can be just `if (auto delay = request.debounce_delay()) { ... }`

Reply:
that seems strictly worse, style wise, which relies on the implicit coversion the semi colon seems easier to read. also we might have a compiler warning against that.